### PR TITLE
replacing styled-components by styletron/baseweb

### PR DESCRIPTION
**Pull request by jckr**
(this is not extensively tested and not ready to land)

This is an attempt to use styletron / baseweb as the main way styles are expressed. if done right this will result in:
- one single source of truth for styling - the theme,
- more concise code,
- it goes to show the craft of uber front end tech

### CODE REVIEW

File: modules/gatsby/package.json [hunk 1]
```diff
@@ -72,16 +73,21 @@
     "prismjs": "^1.15.0",
     "probe.gl": "^3.0.1",
     "prop-types": "^15.6.2",
+    "react": "^16.8.6",
```
Comment (ibgreen) [inline modules/gatsby/package.json]: These were previously moved to `peerDependencies` to let the application control React versions.

File: modules/gatsby/src/templates/doc-n.jsx [hunk 2]
```diff
@@ -20,39 +34,60 @@ export const query = graphql`
   }
 `;
 
-function replaceLinks(props) {
-  const { html } = props.data.docBySlug;
-  const { relativeLinks } = props.pageContext;
+// function replaceLinks(props) {
```
Comment (ibgreen) [inline modules/gatsby/src/templates/doc-n.jsx]: Add comment or cut

**Comment by jckr** [comment]
image

File: modules/gatsby/package.json (hunk 1 above)
Comment (jckr) [reply modules/gatsby/package.json]: reverted to that

File: modules/gatsby/src/templates/doc-n.jsx (hunk 2 above)
Comment (jckr) [reply modules/gatsby/src/templates/doc-n.jsx]: removed

File: modules/gatsby/src/components/layout/layout.jsx [hunk 3]
```diff
@@ -3,7 +3,13 @@
 // each page using the gatsby browser/SSR `wrapPage` callback.
 
 import React from 'react';
+<<<<<<< HEAD
 import {lightThemePrimitives, createTheme} from '../styled/theme';
+=======
+import {createTheme} from 'baseui';
+import {primitives} from '../styled/theme';
+
+>>>>>>> use baseui typography for markdown, replace clan pro by uber move
```
Comment (javidhsueh) [inline modules/gatsby/src/components/layout/layout.jsx]: Resolve conflicts?

**Comment by gergelyke** [comment]
🤗

**Comment by ibgreen** [comment]
@gergelyke 	Unfortunately we had to back out the styletron support due to issues during gatsby build. Perhaps we could team up and make another attempt to get this landed?

**Comment by gergelyke** [comment]
that would be great - what issues did you run into?

**Pull request merged**